### PR TITLE
Improve Mylyn.setup's workingsets task

### PR DESCRIPTION
**Pull request by merks**
Also remove the old API baseline task in favor of the new targlets.based task.

### CODE REVIEW

**Comment by merks** [comment]
@BeckerFrank 

I tried to simplify the working set to use the Repository Predicate which now support a pattern matched against the relative path within the repository.  This avoids problems where the location might look like a regular expression..  I didn't need to exclude anything, but in case you do need that, it's helpful to use an exclusion predicate to exclude the projects in some other working set.

I also got rid of the second set baseline task because you can only really set one...

File: org.eclipse.mylyn.releng/oomph/Mylyn.setup [hunk 1]
```diff
@@ -126,10 +126,6 @@
         </repositoryList>
       </targlet>
     </setupTask>
-    <setupTask
```
Comment (ruspl-afed) [inline org.eclipse.mylyn.releng/oomph/Mylyn.setup:129]: +1

**Review by ruspl-afed — APPROVED** [review]
LGTM.
Even if I don't understand some magic here.

**Comment by ruspl-afed** [comment]
Let's wait for @BeckerFrank , he is The Master of Mylyn.setup

File: org.eclipse.mylyn.releng/oomph/Mylyn.setup [hunk 2]
```diff
@@ -151,6 +147,7 @@
   </setupTask>
   <setupTask
       xsi:type="setup:ResourceCreationTask"
+      id="mylyn.workingsets"
```
Comment (BeckerFrank) [inline org.eclipse.mylyn.releng/oomph/Mylyn.setup:150]: I think that is the wrong line should this not be in line 212 instead of 150

**Review by BeckerFrank — CHANGES_REQUESTED** [review]
Please name the correct setup Task as mylyn.workingsets.
The rest is so that we can merge.

**Comment by merks** [comment]
FYI, note how setting an ID avoids positional references to the task that could be broken by textual editing:

image

**Comment by BeckerFrank** [comment]
> FYI, note how setting an ID avoids positional references to the task that could be broken by textual editing:
> 
> image

Thanks I will go and add some more ID's after we have merged this Request.
The ID mylyn.workingsets  for a Resource Creation Task looks wrong for me.

**Comment by merks** [comment]
Sorry, I thought I force pushed earlier.  Too many irons in the fire.  😢

**Review by BeckerFrank — APPROVED** [review]
LGTM

**Pull request merged**